dap/BESDapResponseBuilder.cc

```diff
@@ -284,24 +272,104 @@ int BESDapResponseBuilder::get_timeout() const
     return d_timeout;
 }
 
+/**
+ * Turn on the alarm. The code will timeout after d_timeout
+ * seconds unless timeout_off() is called first.
+ *
+ * @deprecated
+ */
+void
+BESDapResponseBuilder::timeout_on() const
+{
+#if USE_LOCAL_TIMEOUT_SCHEME
+#ifndef WIN32
+    alarm(d_timeout);
```

Comment:
We have a SignalHandler class that could handle some of this, if not all of this. Wouldn't have to worry about using the macros everywhere. timeout_on would call a method on the SignalHandler instance to turn on timeout, turn off timeout. The underlying code is encapsulated in the SignalHandler class.